tsdb/engine/tsm1/wal_test.go

```diff
@@ -651,6 +652,54 @@ func TestWALSegmentReader_Corrupt(t *testing.T) {
 	}
 }
 
+func TestWALRollSegment(t *testing.T) {
+	dir := MustTempDir()
+	defer os.RemoveAll(dir)
+
+	w := tsm1.NewWAL(dir)
+	if err := w.Open(); err != nil {
+		t.Fatalf("error opening WAL: %v", err)
+	}
+	const segSize = 1024
+	w.SegmentSize = segSize
+
+	values := map[string][]tsm1.Value{
+		"cpu,host=A#!~#value": []tsm1.Value{tsm1.NewValue(1, 1.0)},
+		"cpu,host=B#!~#value": []tsm1.Value{tsm1.NewValue(1, 1.0)},
+		"cpu,host=C#!~#value": []tsm1.Value{tsm1.NewValue(1, 1.0)},
+	}
+	if _, err := w.WriteMulti(values); err != nil {
+		fatal(t, "write points", err)
+	}
+
+	encodeSize := int64(0)
+	if files, err := ioutil.ReadDir(w.Path()); err != nil {
+		fatal(t, "ReadDir", err)
+	} else if len(files) != 1 {
+		t.Fatalf("unexpected segments size %d", len(files))
+	} else {
+		encodeSize = files[0].Size()
```

Comment:
Style nit: I don't think this needs to be wrapped in the `else`, since the other cases cause the test to bail out

Reply:
And if you move it out of the `else`, you could remove `encodeSize := int64(0)` above and define it a single time.

Reply:
done. thanks!